permissions_validators/src/private_blockchain/mod.rs

```diff
@@ -38,3 +38,633 @@ impl<W: WorldTrait> IsGrantAllowed<W> for ProhibitGrant {
         Err("Granting at runtime is prohibited.".to_owned())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    #![allow(clippy::restriction)]
+
+    use std::str::FromStr as _;
+
+    use super::*;
+
+    struct TestEnv {
+        // Alice's Id that owns Gold and Bronze assets.
+        alice_id: AccountId,
+        // Bob's Id that owns Silver asset.
+        bob_id: AccountId,
+        // Carol's Id that owns Bronze asset.
+        carol_id: AccountId,
+        // Gold asset's Id.
+        gold_asset_id: AssetId,
+        // Gold asset definition's Id.
+        gold_asset_definition_id: AssetDefinitionId,
+        // Silver asset's Id.
+        silver_asset_id: AssetId,
+        // Silver asset definition's Id.
+        silver_asset_definition_id: AssetDefinitionId,
+        // Bronze asset's Id.
+        bronze_asset_id: AssetId,
+        // Bronze asset definition's Id.
+        bronze_asset_definition_id: AssetDefinitionId,
+        // Wonderland is a domain that contains Alice and Bob
+        wonderland: (DomainId, Domain),
+        // Denoland is a domain that contains Carol
+        denoland: (DomainId, Domain),
+        // World state view contains wonderland and denoland domains.
+        wsv: WorldStateView<World>,
+    }
+
+    impl TestEnv {
+        fn new() -> Self {
```

Comment:
If you have time, please add doc comments, which explain who has which assets to this associated function, or to the struct (preferably both, preferably using `#[doc(include...]`.

Reply:
I couldn't think of anything better than describing relationships in the comments to the fields of `TestEnv` structure. Comments are prompted when you hover over the field. I wanted to use something like `mermaid` but `rustdoc` doesn't support it. Creating a separate markdown file is probably too much.

Reply:
Sounds good.